File: TDDMicroExercises.ProposedSolution/CSharp/TirePressureMonitoringSystem/Alarm.cs
namespace TDDMicroExercises.OneSolution.TirePressureMonitoringSystem
{
    public class Alarm
    {
        public const double LowPressureTreshold = 17;
        public const double HighPressureTreshold = 21;

        readonly ISensor _sensor;

        bool _alarmOn = false;

        public Alarm() : this(new Sensor())
        {
        }

        public Alarm(ISensor sensor)
        {
            _sensor = sensor;
        }

        public void Check()
        {
            double psiPressureValue = _sensor.PopNextPressurePsiValue();

            if (psiPressureValue < LowPressureTreshold || HighPressureTreshold < psiPressureValue)
            {
                _alarmOn = true;
            }
        }

        public bool AlarmOn
        {
            get { return _alarmOn; }
        }

    }
}

File: TDDMicroExercises.ProposedSolution/Javascript/tire-pressure-monitoring-system/alarm.js

Alarm = function(sensor) {

	this._lowPressureTreshold = 17;
	this._highPressureTreshold = 21;
	this._sensor = sensor;
	this._alarmOn = false;
};

Alarm.Create = function() {
	return new Alarm(new Sensor());
};

Alarm.prototype = {

	check: function () {
		
		var psiPressureValue = this._sensor.popNextPressurePsiValue();

		if (psiPressureValue < this._lowPressureTreshold || this._highPressureTreshold < psiPressureValue)
		{
			this._alarmOn = true;
		}
	},
	
	alarmOn: function () {
		 return this._alarmOn;
	}

};
